src/server/lib/rate-limit/bucket.ts

```diff
@@ -0,0 +1,170 @@
+import type Redis from 'ioredis';
+
+import type {
+  ParsedRateLimitBucket,
+  RateLimitBucketContents,
+  BucketKeys,
+  BucketConfiguration,
+  PipelinedBucketData,
+} from './types';
+
```

Comment:
Could we put some of the detail from the PR description about token bucket as comments here ? Maybe a link to the wikipedia article etc